pydantic/_internal/_generate_schema.py

```diff
@@ -534,14 +534,22 @@ def _generate_dc_field_schema(
         Prepare a DataclassField to represent the parameter/field, of a dataclass
         """
         common_field = self._common_field_schema(name, field_info, decorators)
+
+        validation_alias: str | list[str | int] | list[list[str | int]] | None = common_field['validation_alias']
+        serialization_alias: str | None = common_field['serialization_alias']
+        alias_generator = self.config_wrapper.alias_generator
+        if alias_generator:
+            validation_alias = alias_generator(name)
+            serialization_alias = alias_generator(name)
+
```

Comment:
We can move this change into [_common_field_schema](https://github.com/pydantic/pydantic/blob/0346ddb6a35770007f32815d8e4a179b778e0ef4/pydantic/_internal/_generate_schema.py#L549) but this function is used in `_generate_td_field_schema` as well. So, that's why I keep it in the `_generate_dc_field_schema`

Reply:
I think it may be a good idea to move into `_common_field_schema` since soon there will be something equivalent for `_generate_model_field_schema` (change coming down the pipe in pydantic core). Does it behave horribly/not work for typeddict if someone did add an alias? I think it's reasonable if someone might want to use aliases with typeddict too.

Reply:
I moved it to `_common_field_schema` and also added a test for typeddict inspired by your example.